### ISSUE TITLE
licensing question

### ISSUE THREAD

**Issue by aovasylenko** [xolox/python-deb-pkg-tools#20]
hi @xolox , thank you for building the package and distribution it under the MIT license. We are evaluating it, but found, that you are using `python-debian`, which is distibuted under GPL2 or later. Did you evaluate, how it works together? Thank you in advance

**Comment by xolox** [comment]
Hi @aovasylenko and thanks for the feedback.

To be honest I didn't think this through enough! Prompted by your question I researched the topic and it seems that by licensing deb-pkg-tools as MIT I'm violating the [python-debian](https://salsa.debian.org/python-debian-team/python-debian) GPL2 license. This is rather unfortunate and was never my intention.

So it seems I have two options:

1. Change the license of deb-pkg-tools to GPL2.
2. Remove the python-debian dependency.

Going with the first option isn't necessarily a problem for me because I release what I do as open source software anyway, but I must admit to having a dislike for the viral nature of the GPL. Considering the other option, the only use case for the python-debian dependency is parsing of the deb822 format:

```shell
$ grep -r 'from debian'
deb_pkg_tools/control.py:from debian.deb822 import Deb822
deb_pkg_tools/tests.py:from debian.deb822 import Deb822
deb_pkg_tools/package.py:from debian.deb822 import Deb822
deb_pkg_tools/printer.py:from debian.deb822 import Deb822
```

Given the total scope of deb-pkg-tools this is a rather small bit of functionality, although clearly essential to the whole. Famous last words: It doesn't seem hard to implement deb822 parsing myself based on the [specification](https://manpages.debian.org/deb822) (which is only 690 words).

**Comment by xolox** [comment]
I'm under the impression I may be able to get away with replacing the deb822 parsing based on the python-debian package with [email.parser.HeaderParser](https://docs.python.org/3/library/email.parser.html#email.parser.HeaderParser) from the Python standard library. If anyone is reading along and knows I'm wrong then by all means point that out :innocent:.

**Comment by imanhodjaev** [comment]
@xolox you can also emulate ini file and parse headers using configparser :)

**Comment by xolox** [comment]
Thanks @imanhodjaev for the suggestion, that is indeed another option.

**Comment by xolox** [comment]
**Status update:** I've certainly worked myself into a uncomfortable corner here, because it turns out that deb-pkg-tools integrates with not one but two GPL2 licensed libraries:

- [python-apt](https://pypi.org/project/python-apt/) is used for version comparison.
- [python-debian](https://salsa.debian.org/python-debian-team/python-debian) is used for deb822 parsing.

I can't believe I never realized this before :man_facepalming:. The deb822 parsing seems simple enough to replace but the version comparison is more involved. Of course I can also give in and change the license to GPL2 but this will ripple through the little ecosystem of Python packages that I've developed in the past seven years :disappointed:.

**Comment by imanhodjaev** [comment]
@xolox btw I created my own parser it should work the same way with archive as deb822 did https://github.com/imanhodjaev/pycran

**Comment by xolox** [comment]
I've just released [deb-pkg-tools 8.0](https://deb-pkg-tools.readthedocs.io/en/latest/changelog.html#release-8-0-2020-04-25) which resolves this issue by switching to native (Python) implementations of deb822 parsing and Debian version number comparisons. Thanks for the feedback!
